pos-print/src/main/java/com/clouway/pos/print/core/PrintingListener.kt

```diff
@@ -0,0 +1,16 @@
+package com.clouway.pos.print.core
+
+/**
+ * Provides the methods to give feedback after a print
+ * has been attempted.
+ *
+ * @author Tsvetozar Bonev (tsvetozar.bonev@clouway.com)
+ */
+interface PrintingListener {
+  /**
+   * Notifies that a print response has been created.
+   *
+   * @param printResponse the response to send
+   */
+  fun onReceiptPrinted(receipt: Receipt, printStatus: PrintStatus)
```

Comment:
hm this interface doesn't look required for any functionality

Reply:
I disagree.

Reply:
Where this interface is used in the code, I couldn't find a place

Reply:
It's implemented in the Receipt Repository to avoid bunching up dependencies in the background service.

Reply:
the implementation looks like this : 
```
override fun onReceiptPrinted(receipt: Receipt, printStatus: PrintStatus) {
    val logger = LoggerFactory.getLogger(PrintingListener::class.java)
     logger.info("Receipt was processed")
    logger.info(receipt.receiptId)
    logger.info(printStatus.name)
  }
```
it is placed in the PersistentReceiptRepository, also this method is called from the same class and it is not used anywhere else in the app, that's why I think at this time is not used at all.
In this version of the PR, this interface is not properly used and the code will work and without it.

Reply:
I implemented the interface in a separate class and injected it into the repository to avoid it having to implement both interfaces, but I disagree that the interface serves no purpose as the current logging notification will not be the only effect that will happen if we want to notify the user of the status in the future and thus an interface needs to exist as a grounds for that future feature.